Return-Path: <tiwai@suse.de>
Delivered-To: bero@lindev.ch
Received: from mail.lindev.ch
	by mail.lindev.ch with LMTP
	id iVDLMzXeiFxjLgAABgQX3A
	(envelope-from <tiwai@suse.de>)
	for <bero@lindev.ch>; Wed, 13 Mar 2019 11:40:53 +0100
Received: from mx1.suse.de (mx2.suse.de [195.135.220.15])
	(using TLSv1.2 with cipher AECDH-AES256-SHA (256/256 bits))
	(No client certificate requested)
	by mail.lindev.ch (Postfix) with ESMTPS id 2F7AA3FDD6
	for <bero@lindev.ch>; Wed, 13 Mar 2019 11:40:49 +0100 (CET)
X-Virus-Scanned: by amavisd-new at test-mx.suse.de
Received: from relay2.suse.de (unknown [195.135.220.254])
	by mx1.suse.de (Postfix) with ESMTP id 6F22BAD55;
	Wed, 13 Mar 2019 10:39:26 +0000 (UTC)
Date: Wed, 13 Mar 2019 11:39:26 +0100
Message-ID: <s5ho96fukgx.wl-tiwai@suse.de>
From: Takashi Iwai <tiwai@suse.de>
To: Takashi Iwai <tiwai@suse.de>
Cc: "Bernhard Rosenkraenzer" <bero@lindev.ch>,
	<linux-kernel@vger.kernel.org>,
	<alsa-devel@alsa-project.org>,
	<perex@perex.cz>
Subject: Re: [PATCH] Fix speakers on Acer Predator Helios 500 Ryzen laptops
In-Reply-To: <s5hpnqvukvb.wl-tiwai@suse.de>
References: <20190304233818.GA25749@lindev.ch>
	<s5hpnqvukvb.wl-tiwai@suse.de>
User-Agent: Wanderlust/2.15.9 (Almost Unreal) SEMI/1.14.6 (Maruoka)
 FLIM/1.14.9 (=?UTF-8?B?R29qxY0=?=) APEL/10.8 Emacs/25.3
 (x86_64-suse-linux-gnu) MULE/6.0 (HANACHIRUSATO)
MIME-Version: 1.0 (generated by SEMI 1.14.6 - "Maruoka")
Content-Type: text/plain; charset=US-ASCII

On Wed, 13 Mar 2019 11:30:48 +0100,
Takashi Iwai wrote:
> 
> On Tue, 05 Mar 2019 00:38:19 +0100,
> Bernhard Rosenkraenzer wrote:
> > 
> > On an Acer Predator Helios 500 (Ryzen version), the laptop's speakers
> > don't work out of the box.
> > 
> > The problem can be worked around with hdajackretask, remapping the
> > "Black Headphone, Right side" pin (0x21) to the Internal speaker.
> > 
> > This patch adds a quirk to change this mapping by default.
> > 
> > Signed-off-by: Bernhard Rosenkraenzer <bero@lindev.ch>
> 
> Applied, thanks.

... and looking at the patch again, I found that you added
ALC299_FIXUP_PREDATOR_SPK definition in the wrong place.  It should be
a quirk for alc269-variant while the patch added to alc882-variant.

Also, the addition to alc269_pin_fixup_tbl looks superfluous.  It's a
table to look up the pattern matching of pin configs BIOS sets up, not
the result of the pins after applying the quirk.

The corrected patch is below.  Could you check whether it still works
as expected?


thanks,

Takashi

-- 8< --
From: Bernhard Rosenkraenzer <bero@lindev.ch>
Subject: [PATCH] ALSA: hda/realtek - Fix speakers on Acer Predator Helios 500 Ryzen laptops

On an Acer Predator Helios 500 (Ryzen version), the laptop's speakers
don't work out of the box.

The problem can be worked around with hdajackretask, remapping the
"Black Headphone, Right side" pin (0x21) to the Internal speaker.

This patch adds a quirk to change this mapping by default.

Signed-off-by: Bernhard Rosenkraenzer <bero@lindev.ch>
Cc: <stable@vger.kernel.org>
Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 sound/pci/hda/patch_realtek.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/sound/pci/hda/patch_realtek.c b/sound/pci/hda/patch_realtek.c
index c8413d44973c..d47e2fd0a523 100644
--- a/sound/pci/hda/patch_realtek.c
+++ b/sound/pci/hda/patch_realtek.c
@@ -5673,6 +5673,7 @@ enum {
 	ALC225_FIXUP_HEADSET_JACK,
 	ALC293_FIXUP_SYSTEM76_MIC_NO_PRESENCE,
 	ALC285_FIXUP_LENOVO_PC_BEEP_IN_NOISE,
+	ALC299_FIXUP_PREDATOR_SPK,
 };
 
 static const struct hda_fixup alc269_fixups[] = {
@@ -6639,6 +6640,13 @@ static const struct hda_fixup alc269_fixups[] = {
 		.chained = true,
 		.chain_id = ALC285_FIXUP_LENOVO_HEADPHONE_NOISE
 	},
+	[ALC299_FIXUP_PREDATOR_SPK] = {
+		.type = HDA_FIXUP_PINS,
+		.v.pins = (const struct hda_pintbl[]) {
+			{ 0x21, 0x90170150 }, /* use as headset mic, without its own jack detect */
+			{ }
+		}
+	},
 };
 
 static const struct snd_pci_quirk alc269_fixup_tbl[] = {
@@ -6655,6 +6663,7 @@ static const struct snd_pci_quirk alc269_fixup_tbl[] = {
 	SND_PCI_QUIRK(0x1025, 0x079b, "Acer Aspire V5-573G", ALC282_FIXUP_ASPIRE_V5_PINS),
 	SND_PCI_QUIRK(0x1025, 0x102b, "Acer Aspire C24-860", ALC286_FIXUP_ACER_AIO_MIC_NO_PRESENCE),
 	SND_PCI_QUIRK(0x1025, 0x106d, "Acer Cloudbook 14", ALC283_FIXUP_CHROME_BOOK),
+	SND_PCI_QUIRK(0x1025, 0x1246, "Acer Predator Helios 500", ALC299_FIXUP_PREDATOR_SPK),
 	SND_PCI_QUIRK(0x1025, 0x128f, "Acer Veriton Z6860G", ALC286_FIXUP_ACER_AIO_MIC_NO_PRESENCE),
 	SND_PCI_QUIRK(0x1025, 0x1290, "Acer Veriton Z4860G", ALC286_FIXUP_ACER_AIO_MIC_NO_PRESENCE),
 	SND_PCI_QUIRK(0x1025, 0x1291, "Acer Veriton Z4660G", ALC286_FIXUP_ACER_AIO_MIC_NO_PRESENCE),
@@ -7049,6 +7058,7 @@ static const struct hda_model_fixup alc269_fixup_models[] = {
 	{.id = ALC255_FIXUP_DELL_HEADSET_MIC, .name = "alc255-dell-headset"},
 	{.id = ALC295_FIXUP_HP_X360, .name = "alc295-hp-x360"},
 	{.id = ALC225_FIXUP_HEADSET_JACK, .name = "alc-sense-combo"},
+	{.id = ALC299_FIXUP_PREDATOR_SPK, .name = "predator-spk"},
 	{}
 };
 #define ALC225_STANDARD_PINS \
-- 
2.16.4

